### PR TITLE
EmmyLua: save timestamps of docs.json to skip if no change

### PR DESCRIPTION
**Pull request by muescha**
save timestamps in to a file `timestamps.json` to check if `docs.json` are changed.
if not changed then skip the file check

add also a check to skip the writing to `timestamps.json`

### CODE REVIEW

**Comment by xaviervalarino** [comment]
@muescha happy to see someone else was experiencing startup lag in Hammerspoon caused by EmmyLua uneccessarily reading the jsonDocs files.

I was about to submit a PR for this before  I came across yours. The main difference is that I am storing timestamps in [`hs.settings`](https://www.hammerspoon.org/docs/hs.settings.html) instead of writing it out to a JSON file. I think using `hs.settings` as a serialized store is slightly faster than reading and writing to JSON, but the speed-up is probably negligible. You can see the diff here: https://github.com/xaviervalarino/Spoons/commit/848284ae9362c3d9a53c1a98cef7dc5d6dc9b452

Looking at yours, I do think storing the timestamp in the local settings variable is the way to go.

Hope this PR gets merged, as it significantly speeds up Hammerspoon's startup time when there are no new docs for EmmyLua to generate.

**Comment by muescha** [comment]
My solution don't blow up the internal memory with data, which is maybe unused when he deactive the spoon. 

Storing the timestamps in a file gives the user the control over the metadata. 

A user can not delete the (unknown) meta data stored in `hs.settings`

**Comment by xaviervalarino** [comment]
Thanks for pointing that out. I had not considered having the data hanging around.

File: Source/EmmyLua.spoon/init.lua [hunk 1]
```diff
@@ -183,28 +188,71 @@ function M.create(jsonDocs, prefix)
...
+      M.logger.i("skipping " .. fname)
     end
   end
 end
 
+function M.createWhenChanged(jsonDocs, prefix)
+  local mtime = hs.fs.attributes(jsonDocs, "modification")
+  local timestamp = options.timestamps[jsonDocs]
+
+  if(timestamp == nil or mtime > timestamp) then
```
Comment (muescha) [inline Source/EmmyLua.spoon/init.lua:208]: Maybe better use here `~=` ?
```suggestion
  if(timestamp == nil or mtime ~= timestamp) then
```

for example If I checkout different commits from git then I can also go back and forth in time?

**Comment by muescha** [comment]
@cmsj irc GitHub Action ist hanging

**Comment by pecigonzalo** [comment]
I believe this was ready to merge, could we re-run the CI check?

**Comment by muescha** [comment]
@pecigonzalo I changed some whitespace to re-run CI Check - now all checks have passed :)

**Comment by muescha** [comment]
thx :)

**Pull request merged**